File: src/models/History.js
export class History {
    constructor(systemPrompt) {
        this.messages = [
            { role: 'system', content: systemPrompt }
        ];
    }

    add(role, content, tool_calls = null, tool_call_id = null) {
        const message = { role, content };
        if (tool_calls) message.tool_calls = tool_calls;
        if (tool_call_id) message.tool_call_id = tool_call_id;
        this.messages.push(message);
    }

    get() {
        return this.messages;
    }
}

File: mini_nebulus/models/history.py
from typing import List, Dict, Optional, Any


class History:
    def __init__(self, system_prompt: str):
        self.messages: List[Dict[str, Any]] = [
            {"role": "system", "content": system_prompt}
        ]

    def add(
        self,
        role: str,
        content: Optional[str] = None,
        tool_calls: Optional[List[Dict]] = None,
        tool_call_id: Optional[str] = None,
    ):
        message = {"role": role}
        if content is not None:
            message["content"] = content
        if tool_calls:
            message["tool_calls"] = tool_calls
        if tool_call_id:
            message["tool_call_id"] = tool_call_id

        self.messages.append(message)

    def get(self) -> List[Dict[str, Any]]:
        return self.messages
